### PR TITLE
[Backed Receipt] Download and share a file not a link

### LINKED ISSUE #10625: [Backend Receipts] Set factual version of WC that supports backend generated receipts

### LINKED ISSUE #10632: [Backend Receipts] When clear set expiration days count

### LINKED ISSUE #10648: [Backed Receipt] Download and share a file, not a link

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #10648
Closes: #10625 (now clear that we target 8.7.0 as WC version that will have backend generated receipts)
Closes: #10632 (now clear that 2 days of expiration days will be enough)

### Description

Before this PR we shared the receipt
* Only via email intended
* We attached a link

No, we share in a generic way so it can be sent via any supported app (e.g., messengers), and we share a file, which is downloaded by a given link. File sharing is needed to reduce the amount of issues in cases when a link on a receipt expires.

### Testing instructions

Test on both the release version of WC and the dev version that has a backend receipt generation
* Open a order that has a receipt (all paid orders have receipts if the backend supports receipt generation)
* Click "See receipt" -> Send.
* Notice that you can share it via different apps and that we attach file now

* Collect payment via a card reader
* On the last step click on "send receipt"
* Notice that you can share it via different apps and that we attach file now

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4923871/0a450e9e-976b-454c-b06d-f7503636bc63

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/190">17.2</a>. The due date for this milestone has already passed.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/receipt/PaymentReceiptShareTest.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        on { getExternalFilesDir(anyOrNull()) }.thenReturn(file)
+    }
+
+    private val sut = PaymentReceiptShare(
+        fileUtils = fileUtils,
+        fileDownloader = fileDownloader,
+        context = context,
+    )
+
+    @Test
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/receipt/PaymentReceiptShareTest.kt:32]: nitpick❓ How about adding test that verifies `Success` and `Sharing` state?

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @kidinov. The functionality  works as expected. :shipit:

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/receipt/PaymentReceiptShareTest.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/receipt/PaymentReceiptShareTest.kt:32]: That's not trivial to do so, as we need to hide all android classes and I decided that this the case when effort doesn't worth the impact 🤷‍♂️

**Pull request merged**